Sources/ViewInspector/Inspector.swift

```diff
@@ -1,18 +1,115 @@
+        str = str.replacingOccurrences(of: "<__C.", with: "<")
+
+        Inspector.sanitizedNamespacesCache[typeName] = str
+
+        return str
+
+    }
+
+    /// Replaces the generic types of a given type name with a string.
+    /// Calls to this method are memoized and retained for the lifetime of the program.
+    /// - Parameters:
+    ///   - typeName: The original type name. (e.g. ``SomeTypeName<SomeGenericTypeParameter>``)
+    ///   - replacement: The string to replace the generic parameters with. (e.g. "<EmptyView>")
+    /// - Returns: The type name with its generic parameters replaced. (e.g. ``SomeTypeName<EmptyView>``)
+    static func replaceGenericParameters(inTypeName typeName: String,
+                                         withReplacement replacement: String) -> String {
+        // Check memoized value
+        if let typeNameDict = Inspector.replacedGenericParametersCache[typeName],
+           let cachedResult = typeNameDict[replacement] {
+            return cachedResult
+        }
+
+        // Compute value
+        let result = computeReplacementOfGenericParameters(inTypeName: typeName,
+                                                           withReplacement: replacement)
+
+        // Store memoized value
+        var typeNameDict = Inspector.replacedGenericParametersCache[typeName] ?? [:]
+        typeNameDict[replacement] = result
+        Inspector.replacedGenericParametersCache[typeName] = typeNameDict
+
+        return result
+    }
+
+    private static func computeReplacementOfGenericParameters(inTypeName typeName: String,
+                                                              withReplacement replacement: String) -> String {
+        guard let start = typeName.firstIndex(of: "<") else {
+            return typeName
+        }
+
+        var balance = 1
+        var current = typeName.index(after: start)
+        while balance > 0 && current < typeName.endIndex {
+            let char = typeName[current]
+            if char == "<" { balance += 1 }
+            if char == ">" {
+                guard let indexOfPreviousChar = typeName.index(current,
+                                                               offsetBy: -1,
+                                                               limitedBy: typeName.startIndex) else {
+                    return typeName
+                }
+                let previousChar = typeName[indexOfPreviousChar]
+                if previousChar == "-" {
+                    // We've found the "->" arrow for a closure type. Ignore this ">".
+                } else {
+                    balance -= 1
+                }
+            }
+            current = typeName.index(after: current)
+        }
+
+        if balance == 0 {
+            return String(typeName[..<start]) +
+            replacement +
+            replaceGenericParameters(inTypeName: String(typeName[current...]),
```

Comment:
We still call into the memoized func to potentially leverage from memoized subtypes.

Reply:
Smart!